pkg/runtime/k0s/runtime.go

```diff
@@ -0,0 +1,197 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k0s
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/sealerio/sealer/common"
+	"github.com/sealerio/sealer/pkg/registry"
+	"github.com/sealerio/sealer/pkg/runtime"
+	"github.com/sealerio/sealer/pkg/runtime/k0s/k0sctl"
+	v2 "github.com/sealerio/sealer/types/api/v2"
+	utilsnet "github.com/sealerio/sealer/utils/net"
+	"github.com/sealerio/sealer/utils/platform"
+	"github.com/sealerio/sealer/utils/ssh"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	Vlog      int
+	RegConfig *registry.Config
+	// 	clusterFileK0sConfig is k0sctl cluster config file
+	K0sConfig *k0sctl.K0sConfig
+}
+
+// Runtime struct is the runtime interface for k0s
+type Runtime struct {
+	*sync.Mutex
```

Comment:
I do not see any place where used this Mutex. I am wondering if this is redundant? @starComingup

Reply:
Copy form k8s, now remove it!